packages/forms/src/__tests__/form.test.js

```diff
@@ -150,4 +150,28 @@ describe('Form', () => {
       expect(input).toHaveFocus()
     })
   })
+
+  it('lets users pass custom coercion functions', async () => {
+    const mockFn = jest.fn()
+    const coercionFunction = (value) => parseInt(value.replace('_', ''), 10)
+
+    render(
+      <Form onSubmit={mockFn}>
+        <TextField
+          name="tf"
+          defaultValue="123_456"
+          dataType={coercionFunction}
```

Comment:
I like the flexibility this provides, but I think the prop name should change. It feels weird to pass a function that returns a value (and not a data type) to a prop called "dataType".

My suggestion is to create a new prop, `coercion`, that can take both a string (enum, when we go full TS) and a function. `coercion="Float"` would use the built-in float coercion function, and `coercion={myCoercionFn}` would use your custom function. And then we could add a deprecation warning to the `dataType` prop

Reply:
Thanks @Tobbe, I agree with the renaming of this prop, but I would like to make an alternative suggestion `transformValue`

Reply:
I'm wondering where `dataType` originally came from.

Reply:
Yeah, I wonder who PRed that weird name 🙈 😀 

`transformValue` sounds good to me 🚀

Reply:
@Tobbe It's not a weird name, I just wanted to make sure that we're staying within the intentions of the original author - if that was you then awesome!

Reply:
It was me. And yes, this was my intention all along. Love the change going in here 👍